lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1218,6 +1219,11 @@ static bool performCompileStepsPostSILGen(
   SILOptions &SILOpts = Invocation.getSILOptions();
   IRGenOptions &IRGenOpts = Invocation.getIRGenOptions();
 
+  BufferIndirectlyCausingDiagnosticRAII ricd(
```

Comment:
In case it's useful, here's the idiom for an RAII you may or may not want:

```cxx
Optional<BufferIndirectlyCausingDiagnosticRAII> bicd;
if (condition)
  bicd.emplace(Context.Diags, "foo");
```

Reply:
Thanks, that's better. I 'll use it.